### ISSUE TITLE
Random not found error while trying to read app settings at constructor

### ISSUE THREAD

**Issue by panuoksala** [aspnet/MicrosoftConfigurationBuilders#222]
We are using **Microsoft.Configuration.ConfigurationBuilders.AzureKeyVaultConfigBuilder** in a .NET Framework 4.8 Web API project. Normally everything works, but sometimes (randomly) our application fails to get secret from KeyVault. I think this is related to constructor/static/locking, because it happens mostly at Controller constructor. For example we have a class that is registered into Autofac DI container and that class tries to read connectionstring from KeyVault at constructor. This class is constructured at API Controller constructor.

```
 var builder = new ServiceBusConnectionStringBuilder(ConfigurationManager.AppSettings["ServiceBusConnectionString"])
            {
                EntityPath = entityPath
            };
```

Sometimes that line creates following error:

```
An error occurred when trying to create a controller of type 'InvoiceController'. Make sure that the controller has a 
parameterless public constructor. An exception was thrown while activating API.Controllers.InvoiceController -> 
Common.Invoicing.CreateInvoicePdfService -> Integration.ServiceBus.Services.PaymentService. 
An exception was thrown while invoking the constructor 'Void .ctor()' on type 'PaymentService'. 
Value for the connection string parameter name 'key vault' was not found.
Parameter name: connectionString 
```

Error Call Stacks

```
Inner exception System.Exception handled at System.Configuration.BaseConfigurationRecord.EvaluateOne:
   at Microsoft.Configuration.ConfigurationBuilders.KeyValueConfigBuilder.EnsureGreedyInitialized (Microsoft.Configuration.ConfigurationBuilders.Base, Version=2.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35)
   at Microsoft.Configuration.ConfigurationBuilders.KeyValueConfigBuilder.ProcessConfigurationSection (Microsoft.Configuration.ConfigurationBuilders.Base, Version=2.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35)
   at System.Configuration.BaseConfigurationRecord.CallHostProcessConfigurationSection (System.Configuration, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a)
Inner exception System.NullReferenceException handled at Microsoft.Configuration.ConfigurationBuilders.KeyValueConfigBuilder.EnsureGreedyInitialized:
   at Microsoft.Configuration.ConfigurationBuilders.AzureKeyVaultConfigBuilder.GetAllValues (Microsoft.Configuration.ConfigurationBuilders.Azure, Version=2.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35)
   at Microsoft.Configuration.ConfigurationBuilders.KeyValueConfigBuilder.EnsureGreedyInitialized (Microsoft.Configuration.ConfigurationBuilders.Base, Version=2.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35)
```

Could we call something at startup to ensure that everything is properly loaded, before starting to initialize DI container?

**Comment by StephenMolloy** [comment]
Looks like you're still on the 2.0 config builders. This might be #140, which should be fixed in the 3.0 release.

**Comment by StephenMolloy** [comment]
It's been a while. I assume this has been cleared up with the 3.0 release.
